### ISSUE TITLE
Extension invocation issue

### ISSUE THREAD

**Issue by innosam** [Azure/WALinuxAgent#1170]
There is a change made in guest agent code which will cause site recovery Linux extensions to fail. https://github.com/Azure/WALinuxAgent/commit/3c55507d4672eef1ce9ce72d788c1f274e1fa55b 
-	The change is not released yet.
-	The impact is yet limited to one customer, with RHEL6.8 VM's, who seems to have upgraded the code from the master branch directly. A CRI has been opened by the same customer.
-	I contacted the customer & got his VHD copied locally to reproduce the issue but that didn’t work. However, when I updated the test VM with the Guest Agent master branch directly, the issue repro’ed.

```
-            devnull = open(os.devnull, 'w')
-            child = subprocess.Popen(base_dir + "/" + cmd,  <------
-                                     shell=True,
-                                     cwd=base_dir,
-                                     stdout=devnull,
-                                     env=os.environ)
+            # This should be .run(), but due to the wide variety
+            # of Python versions we must support we must use .communicate().
+            process = subprocess.Popen(os.path.join(base_dir, cmd),   <------
+                                  shell=True,
+                                  cwd=base_dir,
+                                  stdout=subprocess.PIPE,
+                                  stderr=subprocess.PIPE,
+                                  env=os.environ)
```

**Root cause**:
On launching the commands, the agent has started using os.path.join() function instead of creating the path themselves as explained below.
In case of os.path.join() function, the latter strings shouldn't start with a slash. If they start with a slash, then they're considered an "absolute path" and everything before them is discarded.
For Example:
os.path.join('/home/build/test/sandboxes/', '/new_sandbox/')
This will return only ‘/new_sandbox/’ and previous string will be discarded.

**Mitigation**:
We will have to take a fix either in GA/Extension or both. This might break some other extensions as well (this would cause a regression for working code).
-	We will remove the forward slash immediately from the manifest file in our latest payload, which will start shipping next week.
-	However, if GA releases today, the new enable protections for Linux VMs will start breaking for us until we fix the parameter for our extensions & deploy.

**Comment by jasonzio** [comment]
The Sire Recovery agent is incorrect and must be fixed. Specifying a leading slash on cmd was always incorrect, and it was merest happenstance that it worked in the past.

**Comment by jasonzio** [comment]
I have been convinced to change the agent to allow the (incorrect, but historical) behavior, in which an extension could specify the command with a leading slash to be interpreted as relative to the extension directory.

**Comment by innosam** [comment]
Great to know this, I have fixed the issue in the site recovery extension as well.

**Comment by jasonzio** [comment]
Resolved by #1187
